src/main/java/org/veupathdb/service/eda/ds/core/AbstractPlugin.java

```diff
@@ -172,16 +174,18 @@ public final Consumer<OutputStream> processRequest(String appName, T request, En
     _requestProcessed = true;
     logRequestTime("Initial request processing complete");
 
+    // create stream generator
+    Optional<TwoTuple<String, Object>> typedTuple = _computeInfo.map(info -> new TwoTuple<>(info.getFirst(), info.getSecond()));
+    Function<StreamSpec, ResponseFuture> streamGenerator = spec -> _mergingClient
+        .getTabularDataStream(_referenceMetadata, _subsetFilters, _derivedVariableSpecs, typedTuple, spec);
+
+    final AutoCloseableList<InputStream> dataStreams = StreamingDataClient.buildDataStreams(_requiredStreams, streamGenerator);
```

Comment:
It's funny how this makes me very nervous not to have this just sitting "naked" between creation here and the calling of the Consumer<OutputStream>.  But should be safe in theory.